### ISSUE TITLE
RHEL 7 build -- some tests fail.

### ISSUE THREAD

**Issue by pelesh** [GridOPTICS/GridPACK#19]
I tried to build GridPACK from current branch `release/3.2`on RHEL 7 following instructions from [here](https://www.gridpack.org/wiki/index.php/Building_on_RHEL). I used mpich and Boost library from RHEL 7 packages and I built PETSc and GA libraries following the instructions under the link above. To configure GridPACK installation, I used following script:
```
#!/bin/bash
cmake3 -Wdev \
       -D PETSC_DIR:STRING="/usr/local/petsc" \
       -D GA_DIR:STRING="/usr/local/ga" \
       -D USE_PROGRESS_RANKS:BOOL=FALSE \
       -D MPI_CXX_COMPILER:STRING='mpicxx' \
       -D MPI_C_COMPILER:STRING='mpicc' \
       -D MPIEXEC:STRING='mpiexec' \
       -D CMAKE_INSTALL_PREFIX:PATH='/usr/local/gridpack' \
       -D CMAKE_BUILD_TYPE:STRING='RELWITHDEBINFO' \
       -D MPIEXEC_MAX_NUMPROCS:STRING="2" \
       -D CMAKE_VERBOSE_MAKEFILE:BOOL=TRUE \
       ../src
```
Note that I don't need to specify Boost directory, since it is already on the default path. I did, however, have to load mpich module:
```
module load mpi/mpich-x86_64
```
The build went well with a few warnings. Some tests failed however. 
```
The following tests FAILED:
          2 - greetings_parallel (Failed)
         40 - complex_dae_solver_serial (Timeout)
         41 - complex_dae_solver_parallel (Timeout)
         42 - real_dae_solver_serial (Timeout)
         43 - real_dae_solver_parallel (Timeout)
         44 - petsc_ga_matrix_serial (Failed)
         45 - petsc_ga_matrix_parallel (Failed)
         63 - dynamic_simulation_full_y_parallel (Timeout)
         68 - kalman_ds_serial (Timeout)
         69 - kalman_ds_parallel (Failed)
```
Tests 2, 44, and 45 returned error:
```
Required regular expression not found.Regex=[I am process 0 of [1-9].*$
```
DAE examples seem to run correctly, but they take forever to execute, therefore they time out. This is the case for both serial and parallel DAE examples.

**Comment by pelesh** [comment]
I should perhaps add that I had to edit cmake files to add timeouts for `complex_dae_*` and `real_dae_*` tests. Otherwise, these tests run for a really long time. I left them running for several minutes before eventually terminating them. It seems as if there is an infinite loop in those tests.

**Comment by bjpalmer** [comment]
I had some problems with those tests as well when I tried building with shared libraries. They both seemed to hang.

**Comment by wperkins** [comment]
> I should perhaps add that I had to edit cmake files to add timeouts for `complex_dae_*` [...]

You do not need to edit any files to set the test timeout.  When configuring GridPACK, use a `cmake` option something like `-D GRIDPACK_TEST_TIMEOUT:STRING=60`, where the time is in seconds.

**Comment by wperkins** [comment]
I expect the difficulties here are due to some change in PETSc from 3.7 to 3.8. I have no difficulty with these tests on RHEL with PETSc 3.7.6.

**Comment by pelesh** [comment]
> > I should perhaps add that I had to edit cmake files to add timeouts for `complex_dae_*` [...]
> 
> You do not need to edit any files to set the test timeout. When configuring GridPACK, use a `cmake` option something like `-D GRIDPACK_TEST_TIMEOUT:STRING=60`, where the time is in seconds.

It seems timeouts were disabled for these tests. Other tests do have timeouts that are configureable from CMake.

**Comment by wperkins** [comment]
Right. This was fixed in commit [afa3ffc0](https://github.com/GridOPTICS/GridPACK/commit/afa3ffc00c859e25291b710e2a099aec4a5d338b).

**Comment by wperkins** [comment]
I opened #32 to deal with the DAE solver tests.  There are probably other issues with PETSc 3.8.

**Comment by wperkins** [comment]
Closed in favor of #32.
